pages/course/[course]/assignment/[assignment]/submit/index.tsx

```diff
@@ -0,0 +1,104 @@
+import withProtected from "../../../../../../util/withProtected";
+import { queryParamToNumber } from "../../../../../../util/misc";
+import { GetServerSidePropsContext, NextPage } from "next";
+import Badge, { BadgeVariant } from "Components/Badge";
+import Sidebar from "Components/Sidebar";
+import { User, Assignment } from "types";
+import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";
+import Upload from "../../../../../../Components/UploadBox";
+import { useSupabaseClient } from "@supabase/auth-helpers-react";
+import { useState } from "react";
+
+type Submission = {
+  id: string;
+  is_late: boolean;
+  score: number | null;
+  flags: string[] | null;
+  student: User;
+};
+
+type AssignmentT = Assignment & { submission: Submission[] };
+
+interface AssignmentProps {
+  id: number;
+  courseId: number;
+  assignment: AssignmentT;
+}
+
+export const getServerSideProps = (ctx: GetServerSidePropsContext) =>
```

Comment:
Recently found a type `GetServerSideProps` that I like over typing `ctx`- it applies types to the whole function including return types.

ex

```ts
export const getServerSideProps: GetServerSideProps = (ctx) => withProtected(ctx, async (ctx) => {...});
```

Reply:
(not blocking or even necessary to fix)